Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -64,16 +126,103 @@ class PreplannedMapModel: ObservableObject, Identifiable {
         }
     }
     
-    /// A Boolean value indicating if download can be called.
-    var canDownload: Bool {
-        switch status {
-        case .notLoaded, .loading, .loadFailure, .packaging, .packageFailure,
-                .downloading, .downloaded:
-            false
-        case .packaged, .downloadFailure:
-            true
+    /// Updates the status based on the download result of the mobile map package.
+    func updateDownloadStatus(for downloadResult: Optional<Result<MobileMapPackage, any Error>>) {
```

Comment:
This method does more than the doc says and it does more than the name leads one to believe. Namely, it sets the mobileMapPackage property in addition. A method should do a single thing, or it should document everything it does.